### PR TITLE
2567 path app launch

### PR DESCRIPTION
**Pull request by sblack-usu**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case

1.Create a web app resource.  Navigate to the edit page and open the resource specific tab.  

Note the yellow banner at the top with the words ```App Home Page URL or an App-launching URL Pattern```.   

Add an invalid url in the textbox under ```App-launching Aggregation Level URL Pattern``` and click save.  Ensure a Red box notifies you that the url is invalid and the url is not saved.  Do the same for ```App-launching File Level URL Pattern``` and ```App-launching URL Pattern```

Add a valid url using key HS_AGG_PATH, such as ```http://www.my-app-website.org/app_1/?agg=${HS_AGG_PATH}``` under ```App-launching Aggregation Level URL Pattern```.  Click save and ensure the url is saved.  Go Back to view mode, navigate to the resource specific tab and verify the url is listed.  Verify the yellow banner at the top no longer has the words ```App Home Page URL or an App-launching URL Pattern``` in the required message.  Delete the url by saving the textbox when empty.  Verify the yellow banner again has the message  ```App Home Page URL or an App-launching URL Pattern```

Repeat these steps for ```App-launching File Level URL Pattern``` and ```App-launching URL Pattern```

2. Open the web app and enter an invalid File Extension in the textbox ```Supported File Extensions``` (Something like 'garbage').  Save and Validate an error is reported and that the message makes sense.  Now enter a valid comma separated entry for the file extensions which includes .tif files and click save.

Enter a url in the ```App-launching File Level URL Pattern``` that includes the keyword HS_FILE_PATH i.e. ```http://www.my-app-website.org/app_1/?res_id=${HS_FILE_PATH}```.  Click save.  Select Composite Resource under Supported Resource Types and click save.  Go back to view mode and add the webapp to your open with list.

Create a composite resource and add a valid .tif file.  Ensure the composite resource auto-aggregates a Geographic Raster Aggregation. Drill down to the .tif file in the file browser, right click on it and validate the webapp shows up on the menu.  Select the webapp and ensure the url resolves to the url pattern with the path of the .tif file in place of the HS_FILE_PATH key.

3. Open the web app and choose Geographic Raster Aggregation under Supported Aggregation Types, then click Save.

Enter a url in the ```App-launching Aggregation Level URL Pattern``` that includes the keyword HS_AGG_PATH i.e. ```http://www.my-app-website.org/app_1/?res_id=${HS_AGG_PATH}```.  Click save.  Select Composite Resource under Supported Resource Types and click save.  Go back to view mode and add the webapp to your open with list.

Create a composite resource and add a valid .tif file.  Ensure the composite resource auto-aggregates a Geographic Raster Aggregation.  Right click the aggregation and ensure the webapp shows up on the dropdown list.  Select the webapp and ensure the url resolves to the url pattern with the path of the aggregation in place of the HS_AGG_PATH key.

4. Within the composite resource validate the web app does not show up in the open with drop down (if it's the only web app, there will be no open with drop down).  Go to the web app tab, click the web app and validate it opens up the web app resource.  

Within the webapp go to edit mode and click the resource specific tab.  Enter a valid App-Launching URL Pattern (http://www.my-app-website.org/app_1/?res_id=${HS_RES_ID}) and click save.  Go back to the composite resource (don't forget to refresh the page).  Ensure the Open with button appears.  Click on the button and select your webapp, then validate the url opens as expected.

### CODE REVIEW

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by sblack-usu** [comment]
@pkdash, this is ready for you to review when you have some time.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

File: hs_tools_resource/forms.py [hunk 1]
```diff
@@ -45,14 +48,192 @@ class Meta:
...
+        # the order in which the model fields are listed
+        # for the FieldSet is the order these fields will be displayed
+        field_width = 'form-control input-sm'
+        layout = Layout(
+            Field('value', css_class=field_width)
+        )
+        kwargs['element_name_label'] = \
+            "App-launching Aggregation Level URL Pattern <a href='/terms#AppURLPattern' " \
+            "target='_blank'>" \
+            "<font size='3'>Help</font></a>"
```
Comment (aphelionz) [inline hs_tools_resource/forms.py]: what does size="3" resolve to in CSS?

**Comment by hydrobuild** [comment]
Build finished.

**Pull request merged**